diff --git a/ggml/src/ggml-cuda/ggml-cuda.cu b/ggml/src/ggml-cuda/ggml-cuda.cu
--- a/ggml/src/ggml-cuda/ggml-cuda.cu
+++ b/ggml/src/ggml-cuda/ggml-cuda.cu
@@ -3552,7 +3552,7 @@ static void ggml_cuda_op_concat(const struct ggml_ops * self, struct ggml_tensor
             }
             for (int i = 1; i <= concurrent_event->n_streams; ++i) {
                 cudaStream_t stream = cuda_ctx->stream(cuda_ctx->device, i);
-                CUDA_CHECK(hipStreamWaitEvent(stream, concurrent_event->fork_event));
+                CUDA_CHECK(hipStreamWaitEvent(stream, concurrent_event->fork_event, 0));
             }
         }
     };
@@ -3636,7 +3636,7 @@ static void ggml_cuda_op_concat(const struct ggml_ops * self, struct ggml_tensor
                             // Wait on join events of forked streams in the main stream
                             CUDA_CHECK(cudaEventRecord(concurrent_event->join_events[i - 1],
                                                        cuda_ctx->stream(cuda_ctx->device, i)));
-                            CUDA_CHECK(hipStreamWaitEvent(cuda_ctx->stream(), concurrent_event->join_events[i - 1]));
+                            CUDA_CHECK(hipStreamWaitEvent(cuda_ctx->stream(), concurrent_event->join_events[i - 1], 0));
                         }

                         is_concurrent_event_active = false;
